dist/dfu.js

```diff
@@ -85,53 +105,88 @@
         });
     }
 
+    /**
+     * Switch to bootloader/DFU mode by writing to the control point of the DFU Service.
+     * The DFU Controller is not responsible for disconnecting from the application (DFU Target) after the write.
+     * The application (DFU Target) will issue a GAP Disconnect and reset into bootloader/DFU mode.
+     * 
+     * https://infocenter.nordicsemi.com/topic/com.nordic.infocenter.sdk5.v11.0.0/bledfu_appswitching.html?cp=4_0_0_4_1_3_2_2
+     */
     function writeMode(device) {
         return new Promise(function(resolve, reject) {
+            var controlChar = null;
 /*
-            // Disconnect event currently not implemented
+            // Disconnect event currently not implemented...
             device.addEventListener("gattserverdisconnected", () => {
-                log("modeData written");
-                resolve();                
+                log("DFU Target issued GAP Disconnect and reset into Bootloader/DFU mode.");
+                resolve(device);                
             });
 */
             connect(device)
             .then(chars => {
-                log("writing modeData...");
-                chars.controlChar.writeValue(new Uint8Array([1]));
-
-                // Hack to gracefully disconnect without disconnect event
-                setTimeout(() => {
-                    chars.server.disconnect();
-                    setTimeout(() => {
-                        log("modeData written");
-                        resolve(device);
-                    }, 3000);
-                }, 3000);
+                log("enabling notifications");
+                controlChar = chars.controlChar;
+                return controlChar.startNotifications();
+            })
+            .then(() => {
+                log("writing modeData");
+                controlChar.addEventListener('characteristicvaluechanged', handleNotifications);
+                return controlChar.writeValue(new Uint8Array([1, 4]));
+            })
+            .then(() => {
+                log("modeData written");
+                resolve(device); // TODO: once disconnect event is implemented we should resolve in its callback...
             })
             .catch(error => {
                 error = "writeMode error: " + error;
                 log(error);
                 reject(error);
             });
+
+            function handleNotifications(event) {
+                log('received notification on control characteristic - ERROR this should not happen');
+            }
```

Comment:
I don't think there is any point to this actually.. at least I can't think of one. Will remove.